uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/LeaveRequestStatusMatrixTest.php

```diff
@@ -185,4 +258,18 @@ public function allNonPossibleStatusTransitionForLeaveApprover() {
       ['', $leaveRequestStatuses['cancelled']],
     ];
   }
+
+  /**
+   * Return possible status transitions that is only exclusive to the Manager or Admin
+   *
+   * @return array
+   */
+  public function getManagerExclusivePossibleStatusTransitions() {
+    $possibleStaffTransitions = $this->allPossibleStatusTransitionForStaffDataProvider();
+    $possibleManagerTransitions = $this->allPossibleStatusTransitionForLeaveApprover();
+
+    $results = array_diff(array_map('serialize', $possibleManagerTransitions), array_map('serialize', $possibleStaffTransitions));
```

Comment:
Why more suited? If you have to serialize it, it sounds like it might not be the best choice. I don't know of course, if there was some unique property or was to easily compare transitions then it might be cleaner and faster with `array_udiff`. If not then I guess serialization and immediate deserialization is an option, although it seems a bit wasteful.